jqjq.jq

```diff
@@ -2456,35 +2470,60 @@ def fromjqtest:
 def jqjq($args; $env):
   def _parse_args:
     def _f:
+      .[0] as $a |
       if length == 0 then empty
-      elif .[0] == "-h" or .[0] == "--help"       then {help: true}, (.[1:] | _f)
-      elif .[0] == "--jq"                         then {jq: .[1]}, (.[2:] | _f)
-      elif .[0] == "--lex"                        then {lex: true}, (.[1:] | _f)
-      elif .[0] == "--no-builtins"                then {no_builtins: true}, (.[1:] | _f)
-      elif .[0] == "-n" or .[0] == "--null-input" then {null_input: true}, (.[1:] | _f)
-      elif .[0] == "--parse"                      then {parse: true}, (.[1:] | _f)
-      elif .[0] == "--repl"                       then {repl: true}, (.[1:] | _f)
-      elif .[0] == "--run-tests"                  then {run_tests: true}, (.[1:] | _f)
-      elif .[0] == "-s" or .[0] == "--slurp"      then {slurp: true}, (.[1:] | _f)
-      elif .[0] == "--"                           then {filter: .[1]}, (.[2:] | _f)
-      elif .[0] | startswith("-")                 then error("unknown argument: \(.[0])")
-      else {filter: .[0]}, (.[1:] | _f)
+      elif $a == "-h" or $a == "--help"              then {help: true}, (.[1:] | _f)
+      elif $a == "--jq"                              then {jq: .[1]}, (.[2:] | _f)
+      elif $a == "--lex"                             then {lex: true}, (.[1:] | _f)
+      elif $a == "--no-builtins"                     then {no_builtins: true}, (.[1:] | _f)
+      elif $a == "-n" or $a == "--null-input"        then {null_input: true}, (.[1:] | _f)
+      elif $a == "--parse"                           then {parse: true}, (.[1:] | _f)
+      elif $a == "--repl"                            then {repl: true}, (.[1:] | _f)
+      elif $a == "--run-tests"                       then {run_tests: true}, (.[1:] | _f)
+      elif $a == "-s" or $a == "--slurp"             then {slurp: true}, (.[1:] | _f)
+      elif $a == "-C" or $a == "--color-output"      then {color_output: true}, (.[1:] | _f)
+      elif $a == "-M" or $a == "--monochrome-output" then {monochrome_output: true}, (.[1:] | _f)
+      elif $a == "--"                                then {filter: .[1]}, (.[2:] | _f)
+      elif $a | startswith("-")                      then error("unknown argument: \($a)")
+      else {filter: $a}, (.[1:] | _f)
       end;
     ( [_f]
     | add
     );
 
+  # get the ANSI color codes for printing values
+  # corresponds to jv_set_colors and its usage in main
+  def parse_colors($opts; $env):
```

Comment:
fq lacks `JQ_COLORS` support atm (has it's own color config with more stuff) maybe can steal some of this code :)

Reply:
You're welcome to use it for that :). It should represent the C parsing and formatting exactly.

From my tests, it appears the gojq ignores `JQ_COLORS` and it formats differently. Maybe it would be worth coordinating a decision with gojq? I figure you probably inherited that from using gojq as a library.

I submitted https://github.com/jqlang/jq/pull/3034, which fixes some usually-invisible inconsistencies with colors in jq. I think that should be merged regardless, but in addition it might worth bringing jq more into line with gojq. In particular, I don't really like how in jq, `{`, `:`, `,`, and `}` in objects get the object color and `[`, `,`, and `]` in arrays get the array color. I think `:` and `,` punctuation should be colored uniformly across objects and arrays. Personally, I don't see much use in coloring braces and brackets, but many people like coloring them by nesting depth in VS Code, so I can see why this simplistic differentiation could be somewhat useful.

Reply:
> You're welcome to use it for that :). It should represent the C parsing and formatting exactly.

👍 

> From my tests, it appears the gojq ignores `JQ_COLORS` and it formats differently. Maybe it would be worth coordinating a decision with gojq? I figure you probably inherited that from using gojq as a library.

Kind-of, fq uses a modified version of cli/encoder.go from gojq https://github.com/wader/fq/blob/master/internal/colorjson/encoder.go that makes the options non-global and adds some things needed by fq (how to format binary strings etc)

I noticed now that gojq seems to only read `GOJQ_COLORS`, was thinking if i added support in fq i would probably read `FQ_COLORS` and fallback to  `JQ_COLORS`? hmm

> I submitted [jqlang/jq#3034](https://github.com/jqlang/jq/pull/3034), which fixes some usually-invisible inconsistencies with colors in jq. I think that should be merged regardless, but in addition it might worth bringing jq more into line with gojq. In particular, I don't really like how in jq, `{`, `:`, `,`, and `}` in objects get the object color and `[`, `,`, and `]` in arrays get the array color. I think `:` and `,` punctuation should be colored uniformly across objects and arrays. Personally, I don't see much use in coloring braces and brackets, but many people like coloring them by nesting depth in VS Code, so I can see why this simplistic differentiation could be somewhat useful.

I see, hmm haven't really thought about it much so no string opinion, but making them more consistent make sense i think. Let's see what the other maintainers think